homeassistant/components/mailbox/__init__.py

```diff
@@ -86,6 +87,8 @@ async def async_setup_platform(
         component = EntityComponent[MailboxEntity](
             logging.getLogger(__name__), DOMAIN, hass, SCAN_INTERVAL
         )
+        # pylint: disable-next=[protected-access]
+        hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STOP, component._async_shutdown)
```

Comment:
Instead of this, subclass `EntityComponent` and override its `async_setup` so it only sets up the stop listener?

Reply:
Done - but I wonder now if #90938 would not be a better alternative.